### PR TITLE
Java17

### PR DESCRIPTION
**Pull request by benoitgravitee**
**Issue**

https://github.com/gravitee-io/issues/issues/archi-222

**Description**

Align with rest of stack to work with java17

**Additional context**
  * Please look on a commit per commit basis to avoid reformatting ones.
  * I will squash them all and create a BREAKING CHANGE so that we have a 3.x version for users of 4.2 and above

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `3.0.0-java17-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/kubernetes/gravitee-kubernetes/3.0.0-java17-SNAPSHOT/gravitee-kubernetes-3.0.0-java17-SNAPSHOT.zip)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -71,16 +69,16 @@
...
+<!--            <dependency>-->
+<!--                <groupId>io.gravitee.gateway.handlers</groupId>-->
+<!--                <artifactId>gravitee-gateway-handlers-api</artifactId>-->
+<!--                <version>${gravitee-gateway.version}</version>-->
+<!--            </dependency>-->
+<!--            <dependency>-->
+<!--                <groupId>io.gravitee.gateway</groupId>-->
+<!--                <artifactId>gravitee-gateway-reactor</artifactId>-->
+<!--                <version>${gravitee-gateway.version}</version>-->
+<!--            </dependency>-->
```
Comment (remibaptistegio) [inline pom.xml:81]: Forgot to delete this part ?

File: gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/config/KubernetesConfigTest.java [hunk 2]
```diff
@@ -29,18 +29,6 @@
...
-    void should_load_default() {
-        KubernetesConfig instance = KubernetesConfig.getInstance();
-        assertThat(instance.getApiTimeout()).isPositive();
-        assertThat(instance.getWebsocketTimeout()).isPositive();
-        assertThat(instance.getApiVersion()).isEqualTo("v1");
-        assertThat(instance.verifyHost()).isTrue();
-        assertThat(instance.useSSL()).isTrue();
-
-        assertThat(instance).isSameAs(KubernetesConfig.getInstance());
-    }
```
Comment (remibaptistegio) [inline gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/config/KubernetesConfigTest.java:42]: Why did you delete this test ?

**Review by remibaptistegio — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (benoitgravitee) [reply pom.xml:81]: Yep it was clean up.. forgot to remove. THanks

File: gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/config/KubernetesConfigTest.java (hunk 2 above)
Comment (benoitgravitee) [reply gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/config/KubernetesConfigTest.java:42]: It is using a shared instance of configuration. Ideally I'd need execute this test first before all else. As you already know I can't do that. Other tests are modifying this instance, so there is no way I can have a stable build. I already test default value in another test. I just wanted to isolate it, but yeah... so static.

**Comment by phiz71** [comment]
Since the parent is now 22 (JDK 17) and there is a lot of changes in the pom.xml, I would recommend to add a BREAKING CHANGE mention in the relevant commit.

**Review by remibaptistegio — APPROVED** [review]
